### ISSUE TITLE
step number 8

### ISSUE THREAD

**Issue by hedudelgado** [makersacademy/unixoid-challenge#46]
8 Output the count of the variables that begin with T to a new file my/public/files/t-vars.count, e.g. "Overall count: 5" (hint: you'll need to use 'command substitution' in bash)

this is the command I used 
echo ${!T*} | wc -l >> t-vars.count

Basically it prints the number of variables lines that begins with T to t-vars.count, and therefore count it. 
I also tried the command substitution, I may have used it wrong.. I do not know..

but still.. I am not sure what the problem is. and even when everything else is ok, number 8 is still wrong.

Please enter your Github username:
hedudelgado
Please enter your Github password:
Submitting...
Submitted!

Almost there! Feel free to try again and re-submit

The following answers need to be corrected:

          8

any clue about what the problem is? thank you! :)

**Comment by markcmhill** [comment]
It seems to return the variables in one line, so you need to count the number of words not the number of lines?

**Comment by hedudelgado** [comment]
You are right.. 
I already tried this---> echo ${!T*} | wc -w >> t-vars.count
it counts de words, but still... I have the same problem :( 


Almost there! Feel free to try again and re-submit

The following answers need to be corrected:

          8

**Comment by markcmhill** [comment]
Do you have the "Overall count:" part in the file?
You are appending the number to the end of the file, what is already in it?

**Comment by ggwc82** [comment]
"Overall count: " needs to be in the echo command.

>> I think appends to the file as a new line, so better to use > and write to the file in one go

**Comment by ggwc82** [comment]
this task I had to google what command substitution was. this link helped me obtain my answer: http://mywiki.wooledge.org/CommandSubstitution

also, you probably should checkout the git issue for question 8 here: https://github.com/makersacademy/pre_course/issues

For the answer I submitted, I did it in two steps using command substitution, as per the first link. You first want to assign your env command like this:

variable1=$(env....etc)

then for the second step, you want to use echo and write that substitution to your file like this:
echo "Overall count: $variable1" > t-vars.count

That solution uses command substitution, as per the requirement. Hope this helps!
